docs/source/common/lightning_cli.rst

```diff
@@ -511,31 +511,41 @@ There is also the possibility of selecting among multiple classes by giving them
 In this case in the config the :code:`optimizer` group instead of having directly init settings, it should specify
 :code:`class_path` and optionally :code:`init_args`. Sub-classes of the classes in the tuple would also be accepted.
 
-For more complex cases in which there could more than one optimizer and/or learning rate scheduler, the automatic
-implementation of :code:`configure_optimizers` can be disabled by linking the configuration group. An example is:
+The automatic implementation of :code:`configure_optimizers` can be disabled by linking the configuration group. An
+example can be :code:`ReduceLROnPlateau` which requires to specify a monitor. This would be:
 
 .. testcode::
 
     from pytorch_lightning.utilities.cli import instantiate_class, LightningCLI
 
     class MyModel(LightningModule):
 
-        def __init__(self, optimizer1_init: dict):
+        def __init__(self, optimizer_init: dict, lr_scheduler_init: dict):
             super().__init__()
-            self.optimizer1_init = optimizer1_init
+            self.optimizer_init = optimizer_init
+            self.lr_scheduler_init = lr_scheduler_init
 
         def configure_optimizers(self):
-            return instantiate_class(self.parameters(), self.optimizer1_init)
+            optimizer = instantiate_class(self.parameters(), self.optimizer_init)
+            scheduler = instantiate_class(optimizer, self.lr_scheduler_init)
```

Comment:
This has the caveat that your model needs to know about the CLI - which hasn't been the case at all until now.

Just pointing it out.

Reply:
I think it is a reasonable compromise in this case, given how powerful it is.  Also, presumably other non Lightning CLI initialization could use the same approach with `optimizer_init` if they really wanted to.

Reply:
> This has the caveat that your model needs to know about the CLI - which hasn't been the case at all until now.

Yes, I would have preferred to avoid this. But I didn't see any other way. If these classes were more composable instead of requiring params and optimizer to instantiate it could have been done in a cleaner way.

Though now that you mention it, at it is now `instantiate_class` depends on jsonargparse being installed. There is no reason why a model should depend on this. I will change it to remove this dependency.